uportal-war/src/main/java/org/jasig/portal/groups/pags/dao/EntityPersonAttributesGroupStore.java

```diff
@@ -354,20 +350,37 @@ private IPersonTester initializeTester(IPersonAttributesGroupTestDefinition test
         }
    }
 
-    private Set<IEntityGroup> getContainingGroups(String name, Set<IEntityGroup> groups) throws GroupsException
-    {
+    private Set<IEntityGroup> getContainingGroups(String name, Set<IEntityGroup> groups) throws GroupsException {
         logger.debug("Looking up containing groups for {}", name);
-        Set<IPersonAttributesGroupDefinition> pagsGroups = personAttributesGroupDefinitionDao.getPersonAttributesGroupDefinitionByName(name);
-        IPersonAttributesGroupDefinition pagsGroup = pagsGroups.iterator().next();
+        IPersonAttributesGroupDefinition pagsGroup = getPagsGroupDefByName(name);
         Set<IPersonAttributesGroupDefinition> pagsParentGroups = personAttributesGroupDefinitionDao.getParentPersonAttributesGroupDefinitions(pagsGroup);
-        for (IPersonAttributesGroupDefinition parent : pagsParentGroups) {
+        for (IPersonAttributesGroupDefinition pagsParent : pagsParentGroups) {
+            IEntityGroup parent = convertPagsGroupToEntity(pagsParent);
             if (!groups.contains(parent)) {
-                groups.add(convertPagsGroupToEntity(parent));
-                getContainingGroups(parent.getName(), groups);
+                groups.add(parent);
+                getContainingGroups(pagsParent.getName(), groups);
             } else {
-                throw new RuntimeException("Recursive grouping detected! for " + name + " and parent " + parent.getName());
+                throw new RuntimeException("Recursive grouping detected! for " + name + " and parent " + pagsParent.getName());
             }
         }
         return groups;
     }
-}
\ No newline at end of file
+
+    /**
+     * Retrieve an implementation of {@code IPersonAttributesGroupDefinition} with the given {@code name} from
+     * the JPA DAO. There are two assumptions. First, that the DAO handles caching, so caching is not implemented here.
+     * Second, that group names are unique. A warning will be logged if more than one group is found with the same name.
+     *
+     * @param name      group name used to search for group definition
+     * @return          {@code IPersonAttributesGroupDefinition} of named group or null
+     * @see             IPersonAttributesGroupDefinitionDao#getPersonAttributesGroupDefinitionByName(String)
+     * @see             IPersonAttributesGroupDefinition
+     */
+    private IPersonAttributesGroupDefinition getPagsGroupDefByName(String name) {
+        Set<IPersonAttributesGroupDefinition> pagsGroups = personAttributesGroupDefinitionDao.getPersonAttributesGroupDefinitionByName(name);
+        if (pagsGroups.size() > 1) {
+            logger.warn("More than one PAGS group with name {} found.", name);
```

Comment:
`warn` seems insufficiently dire for a violation of a fundamental assumption of the groups system.  `error` at the least?

I'm not sure returning what is effectively a random selection among the illegally multiple groups with the same name is the right behavior.  I think I'd rather it fail with an exception.  Not an exception that results in the entire portal falling over dead, of course, but an exception that at least fails the specific group operation being computed.  Multiple groups with the same name is, alas, not okay.